solana-labs/wallet-adapter
# Encourage developers to hand-pick the selection of wallets most appropriate for their audience

# Problem

As third-party wallet offerings evolve, our starter apps continue to put forth a starter list of wallets that does not. Besides going stale over time, the decision of which wallets a dApp should support is not a responsibility of `wallet-adapter` itself; ultimately, the decision of which wallets to support rests with the dApp publisher themselves.

# Summary of changes

* Create a `FakeWalletAdapter` that conforms to the wallet adapter interface, but otherwise offers no useful signing/sending services.
* Include `FakeWalletAdapter` in all of the starter packs, along with instructions on where to find real wallet adapters.

# Test plan

1. `yarn && yarn build`
2. `cd packages/starter/example && yarn dev`
3. Open https://localhost:3000

https://user-images.githubusercontent.com/13243/182485005-84a20666-7fe3-471a-afa7-d8dedfc5d44a.mov

1. `yarn && yarn build`
2. `cd packages/starter/react-ui-starter && yarn start`
3. Open https://localhost:1234

https://user-images.githubusercontent.com/13243/182486068-c0b2c255-5810-40e4-b001-495cd90e014f.mov

Observe warning on the developer console that links to [this document](https://github.com/solana-labs/wallet-adapter#usage).

<img width="525" alt="image" src="https://user-images.githubusercontent.com/13243/182484831-15a9b15a-b7e4-4a0a-b85f-a8aa77b2ed99.png">
